### ISSUE TITLE
Caffe numerical instability when running the fine-tuning notebook in mac?

### ISSUE THREAD

**Issue by mary-alegro** [BVLC/caffe#5107]
Hi! 
I was testing the notebook **02-fine-tuning** that came with caffe and noticed a weird behaviour in my mac installation: I have to set the learning rate to **1e-46** when running the end-to-end training, otherwise it'll diverge. If I run the same notebook in my Linux installation I only need to set it to 0.0001.

On another test, if I run Google's Deep Dream notebook in my mac Caffe I randomly get `Inf`s during the forward pass. I.e. my forward pass will always go to infinity at some point, but it's not always during the same iteration. The same doesn't happen in my Linux installation. 

My mac installation was compiled in El Captain with Anaconda, Boost 1.61 and Apple's BLAS. All tests run OK during compile time.

Does anybody know what is causing this and how to fix it? 

Thank you!

**Comment by williford** [comment]
Can you also try with the Atlas BLAS? Do you use CUDA? CUDNN?

**Comment by mary-alegro** [comment]
So, I have a version of Atlas compiled from scratch in my system. However I can't make caffe to link against it, no matter what I do. It always finds vecLib. I've tried both the standard way editing Makefile.conf and using cmake. Would you have any suggestion on how to get rid of vecLib?

**Comment by mary-alegro** [comment]
Just for the record, I was able to link Caffe against ATLAS after changing the Makefile. Also, I was able to get Caffe working properly in a fresh OS X installation using a Anaconda and OSX ATLAS. I installed OpenCV with Brew and used Anaconda's HDF5.
